### ISSUE TITLE
using % sign in form parameter leads to false positive.

### ISSUE THREAD

**Issue by bjdijk** [SpiderLabs/ModSecurity#1456]
We use a % sign as a wildcard to search in our (oracle) based app. If I for instance search for an order which looks like %12345%, the follow formdata is passed

EMPTY=&savedquery=102894692519582459452643201754&ORDER_NUMBER=%2512345%25&ORDER_SEGMENT=ORDER&MASTER=&DESCRIPTION=&STATUS=

If I disable rule SecRuleRemoveById 920271, it works. In the detailed debug logging I see the below:

[Tue Jun 13 15:46:30 2017] [error] [client 83.80.29.199] ModSecurity: Warning. Found 1 byte(s) in ARGS:ORDER_NUMBER outside range: 9,10,13,32-126,128-255. [file "/etc/httpd/owasp-modsecurity
-crs/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "1282"] [id "920271"] [rev "2"] [msg "Invalid character in request (non printable characters)"] [severity "CRITICAL"] [ver "OWASP_CRS
/3.0.0"] [maturity "9"] [accuracy "9"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "OWASP_CRS/PROTOCOL_VIOLATION/EVASION"] [tag "para
noia-level/2"] [hostname "xxxx"] [uri "uri"] [unique_id "WT-stgoKaRUAADG9HrAAAAAG"]

It seems that the parameters is not decoded correctly

[13/Jun/2017:15:51:59 +0200] [xxxx/sid#7fca2aa08038][rid#7fca2bbf58c8][uri][4] Operator completed in 7 usec.
[13/Jun/2017:15:51:59 +0200] [xxxx/sid#7fca2aa08038][rid#7fca2bbf58c8][uri][9] T (0) Utf8toUnicode: "%12345%"
[13/Jun/2017:15:51:59 +0200] [xxxx/sid#7fca2aa08038][rid#7fca2bbf58c8][uri][9] T (0) urlDecodeUni: "\x12345%"
[13/Jun/2017:15:51:59 +0200] [xxxx/sid#7fca2aa08038][rid#7fca2bbf58c8][uri][4] Transformation completed in 28 usec.
[13/Jun/2017:15:51:59 +0200] [xxxx/sid#7fca2aa08038][rid#7fca2bbf58c8][uri][4] Executing operator "rx" with param "(^[\"'`;]+|[\"'`]+$)" against ARGS:ORDER_NUMBER.

**Comment by bostrt** [comment]
I copied the request parameters you shared and ran a request with CRS 3.0.0 installed and the request was accepted. I also saw the `"Utf8toUnicode: "%12345%"` mentioned in the debug logs; perhaps that is a red herring. Note that rule [920271](https://github.com/SpiderLabs/owasp-modsecurity-crs/blob/v3.0/master/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf#L1280-L1299) also checks against the request headers and request URI so if something is amok in there, it could also trigger the rule. 

A couple other suggestions:

- Try using the latest CRS
- Open issue with the ModSecurity CRS team : https://github.com/SpiderLabs/owasp-modsecurity-crs/issues/new
- Provide the full request by either capturing mod_dumpio output or reproduce using the curl command (with -v option so others can see the full request)
